bundles/pcmjava/tools.vitruv.applications.pcmjava.pojotransformations.java2pcm/src/tools/vitruv/applications/pcmjava/pojotransformations/java2pcm/Java2PcmClassifier.reactions

```diff
@@ -172,26 +173,35 @@ routine addcorrespondenceAndUpdateRepository(pcm::ImplementationComponentType pc
 //Interface
 reaction InterfaceCreated {
 	after element java::Interface inserted in java::CompilationUnit[classifiers]
-	call createPCMInterface(newValue, affectedEObject)		
+	call createOrFindPCMInterface(newValue, affectedEObject)
 }
 
-routine createPCMInterface(java::Interface javaInterface, java::CompilationUnit compilationUnit) {
+routine createOrFindPCMInterface(java::Interface javaInterface, java::CompilationUnit compilationUnit) {
 	match {
-		val pcmRepository = retrieve pcm::Repository corresponding to ContainersPackage.Literals.PACKAGE
-		val contractsPackage = retrieve java::Package corresponding to pcmRepository tagged with "contracts" 
+		val containingPackage = retrieve java::Package corresponding to ContainersPackage.Literals.PACKAGE
+			with containingPackage.compilationUnits.map[name].contains(compilationUnit.name)
```

Comment:
Does that work if two interfaces with the same name appear in two different packages (i.e. the `name` is equal, but `namespace` is different)? So the namespace might also need to be matched?

Reply:
No, that would not work. Good point! I will change this to include the namespace matching!

Reply:
I also match the namespace:
```java
with containingPackage.compilationUnits.map[namespacesAsString + '.' + name].contains(compilationUnit.namespacesAsString + '.' + compilationUnit.name)
```

Reply:
Perfect, thanks!